pandas/core/ops.py

```diff
@@ -96,6 +96,26 @@ def rxor(left, right):
 
 # -----------------------------------------------------------------------------
 
+def make_invalid_op(name):
+    """
+    Return a binary method that always raises a TypeError.
+
+    Parameters
+    ----------
+    name : str
+
+    Returns
+    -------
+    invalid_op : function
+    """
+    def invalid_op(self, other=None):
+        raise TypeError("cannot perform {name} with this index type: "
```

Comment:
you might need to parameterize this on the work 'index' but not sure (IOW can it be a Series)? or maybe just remove the word index ?

Reply:
ATM it is only used for indexes.  I could add an assertion that self is an Index so that we know to change it if/when the wording ceases to be accurate.